test/unit/test_host.rb

```diff
@@ -65,7 +65,6 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com') == Host.new('example.com')
```

Comment:
@cshaffer I just wanted to understand about the reason for the `==` test deletion. Since `equal?`, `eql?` and and `==` share the same implementation, I think testing all 3 methods is reasonable. 

I noticed in the commit you mentioned this was removed due to a rubocop warning that this 'compares something with itself', but I don't understand why comparing 2 instances of an object is undesirable.